keanu-project/src/main/java/io/improbable/keanu/algorithms/mcmc/nuts/NUTS.java

```diff
@@ -151,57 +154,142 @@ public Statistics getStatistics() {
-            this.adaptCount = adaptCount;
+        /**
+         * @param targetAcceptanceProb The targeted acceptance rate for the step size adaption. This defaults to 0.8.
+         * @return the builder for NUTS
+         */
+        public NUTSBuilder targetAcceptanceProb(double targetAcceptanceProb) {
+            if (targetAcceptanceProb > 1.0 || targetAcceptanceProb < 0) {
+                throw new IllegalArgumentException("Target acceptance probability must be between 0.0 and 1.");
+            }
+            this.targetAcceptanceProb = targetAcceptanceProb;
             return this;
         }
 
-        public NUTSBuilder targetAcceptanceProb(double targetAcceptanceProb) {
-            this.targetAcceptanceProb = targetAcceptanceProb;
+        /**
+         * @param adaptEnabled Set to true if step size adaption is wanted.
+         * @return the builder for NUTS
+         */
+        public NUTSBuilder adaptStepSizeEnabled(boolean adaptEnabled) {
+            this.adaptStepSizeEnabled = adaptEnabled;
             return this;
         }
 
-        public NUTSBuilder adaptEnabled(boolean adaptEnabled) {
-            this.adaptEnabled = adaptEnabled;
+        /**
+         * @param adaptCount The number of samples for which the step size and potential will be tuned. For the
+         *                   remaining samples in which it is not tuned, the step size will be frozen to its last
+         *                   calculated value.
+         * @return the builder for NUTS
+         */
+        public NUTSBuilder adaptCount(int adaptCount) {
+            if (adaptCount < 0) {
+                throw new IllegalArgumentException("Adapt count must be greater than or equal to 0");
+            }
+            this.adaptCount = adaptCount;
             return this;
         }
 
+        /**
+         * @param initialStepSize The initial step size. If this is null then a step size will be calculated using
+         *                        heuristics.
+         * @return the builder for NUTS
+         */
         public NUTSBuilder initialStepSize(Double initialStepSize) {
+            if (initialStepSize <= 0) {
+                throw new IllegalArgumentException("Initial step size must be greater than 0");
+            }
             this.initialStepSize = initialStepSize;
             return this;
         }
 
+        /**
+         * @param adaptPotentialEnabled Set to true if adapting of the mass matrix is wanted.
+         * @return the builder for NUTS
+         */
+        public NUTSBuilder adaptPotentialEnabled(boolean adaptPotentialEnabled) {
+            this.adaptPotentialEnabled = adaptPotentialEnabled;
+            return this;
+        }
+
+        /**
+         * @param potentialAdaptWindowSize The window size for adapting the mass matrix.
+         * @return the builder for NUTS
+         */
+        public NUTSBuilder potentialAdaptWindowSize(int potentialAdaptWindowSize) {
```

Comment:
Probably want to make sure this is ` > 1` otherwise this will just be setting the variance to zero each sample step